qiskit_experiments/database_service/db_analysis_result.py

```diff
@@ -161,19 +153,34 @@ def save(self) -> None:
                 "Analysis result cannot be saved because no experiment service is available."
             )
             return
+        # Get DB fit data
+        value = self.value
+        result_data = {
+            "_value": json.loads(json.dumps(value, cls=self._json_encoder)),
+            "_extra": json.loads(json.dumps(self.extra, cls=self._json_encoder)),
+            "_source": self._source,
+        }
 
-        _result_data = json.loads(json.dumps(self._result_data, cls=self._json_encoder))
-        _result_data["_source"] = self._source
+        # Display compatible float values in in DB
+        if isinstance(value, (int, float, bool)):
+            result_data["value"] = float(value)
```

Comment:
I would rather fix the display in the UI than artificial cast a bool to a float.